vignettes/imcRtools.Rmd

```diff
@@ -418,10 +418,89 @@ plotSpatial(pancreasSCE,
+interactions between cell labels (e.g. cell-types) within grouping levels (e.g.
+images). For full descriptions of the testing approaches, please refer to
+[Shapiro et al., Nature
+Methods](https://www.nature.com/articles/nmeth.4391)[@Shapiro2017] and [Schulz
+et al., Cell
+Systems](https://www.sciencedirect.com/science/article/pii/S2405471217305434)
+[@Schulz2018]
+
+The `imcRtools` package exports the `summarizeNeighborhood` and
+`neighborhoodPermTest` function, which summarize all cell-cell interactions per
+grouping level (e.g. image). As a result, a table is returned where each row
+represents one of all possible cell-type/cell-type interactions among all
+grouping levels. Missing entries or `NA`s indicate missing cell-type labels for
+this grouping level. The next section gives details on how interactions are 
+summarized.
+
+### Summarizing interactions
+
+The `summarizeNeighborhood` function counts the number of edges (interactions)
+between each set of unique cell labels per grouping level.
+Simplified, it counts for each cell of type A the number of neighbors of type B.
+This count is averaged within each unique grouping level (e.g. image) 
+in three different ways:
+
+1. `method = "classic"`: The count is divided by the total number of 
+cells of type A. The final count can be interpreted as "How many neighbors 
+of type B does a cell of type A have on average?" 
+  
+2. `method = "classic"`: The count is divided by the number of cells
+of type A that have at least one neighbor of type B. The final count can be 
+interpreted as "How many many neighbors of type B has a cell of type A on 
+average, given it has at least one neighbor of type B?"
+ 
+3. `method = "patch"`: For each cell, the count is binarized to 0 
+(less than `patch_size` neighbors of type B) or 1 (more or equal to 
+`patch_size` neighbors of type B). The binarized counts are averaged 
+across all cells of type A. The final count can be interpreted as "What 
+fraction of cells of type A have at least a given number of neighbors of 
+type B?"
+
+The `summarizeNeighborhood` returns a `DataFrame` containing the summarized
+counts (`ct`) for all combinations of `from_label`, `to_label` and `group_by`.
+
+```{r summarizeNeighborhood}
+out <- summarizeNeighborhood(pancreasSCE,
+                             group_by = "ImageNb",
+                             label = "CellType",
+                             method = "classic",
+                             colPairName = "knn_interaction_graph")
+out
+```
 
-## Patch detection
+### Testing for significance
+
+In the next instance, one can test if the obtained count is larger or smaller
+compared to what is expected from a random distribution of cell labels. For
+this, the `neighborhoodPermTest` function permutes the labels `iter` times and
+counts interactions as described above. This approach generates a distribution
+of the interaction count under spatial randomness. The observed interaction
```

Comment:
Maybe clarify that it is not complete spatial randomness but random permutation of labels in given spatial structure of cells.